src/compiler/scanner/index.ts

```diff
@@ -181,7 +181,8 @@ export class Scanner {
         case CharacterCodes._7:
         case CharacterCodes._8:
         case CharacterCodes._9:
-          let isFloat: boolean = false
+        case CharacterCodes.dot:
```

Comment:
I don't think we really need starting dot for floating numbering... Any practical usage or strong reason?